Replicator/c4Replicator.hh

```diff
@@ -72,8 +78,37 @@ struct C4Replicator : public RefCounted,
 
     void setSuspended(bool suspended) {
         LOCK(_mutex);
-        if (!setStatusFlag(kC4Suspended, suspended))
+        if(_status.level == kC4Stopped) {
+            // Suspending a stopped replicator?  Get outta here...
+            logInfo("Ignoring a suspend call on a stopped replicator...");
+            return;
+        }
+        
+        if(_status.level == kC4Stopping && !statusFlag(kC4Suspended)) {
+            // CBL-722: Stop was already called or Replicator is stopped,
+            // making suspending meaningless (stop() should override any
+            // suspending or unsuspending)
+            logInfo("Ignoring a suspend call on a stopping replicator...");
+            return;
+        }
+        
+        if(_status.level == kC4Stopping) {
+            // CBL-729: At this point, the suspended state has changed from a previous
+            // call that caused a suspension to start.  Register to restart later
+            // (or cancel the later restart) and move on
+            _cancelStop = !suspended;
+            if(_cancelStop) {
+                logInfo("Request to unsuspend, but Replicator is already suspending.  Will restart after suspending process is completed.");
+            } else {
+                logInfo("Replicator suspension process being spammed (request to suspend followed by at least one request to unsuspend and then suspend again), attempting to cancel restart.");
+            }
+            return;
+        } else if (!setStatusFlag(kC4Suspended, suspended)) {
```

Comment:
why the 'else'?

Reply:
Uncovered during the suspend, unsuspend, suspend exercise.  Suspend will set the suspended flag, but unless this is an `else` then unsuspend will clear it, resulting in the next call to suspend returning early because it thinks `stop` was called because of the lack of suspended flag.

Reply:
Oh, you simply meant that I put in an `else` after a `return`?  Good catch, I don't like doing that either!